### PR TITLE
Add JORA fine-tuning notebook

### PR DESCRIPTION
**Pull request by kinarr**
- Leverages [JORA](https://anique.org/JORA/) to fine-tune Gemma for RAG applications
- Instruction Tuning Dataset: [yahma/alpaca-cleaned](https://huggingface.co/datasets/yahma/alpaca-cleaned) which is a subset of the original [Stanford Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset
- Models:
   - [Gemma 2 2B IT Flax model](https://www.kaggle.com/models/google/gemma-2/Flax/gemma2-2b-it) from Kaggle 
   - [Gemma 2 2B IT model](https://huggingface.co/google/gemma-2-2b-it) from Hugging Face

Update: Switched from Gemma 1.1 to 2

### CODE REVIEW

File: Gemma/Finetune_with_JORA.ipynb [hunk 1]
```diff
@@ -0,0 +1,802 @@
+{
```
Comment (windmaple) [inline Gemma/Finetune_with_JORA.ipynb:1]: <p>Not sure why RAG is relevant here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/100/discussion/#comment-1851447740'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (windmaple) [inline Gemma/Finetune_with_JORA.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    VARIANT = "1.1-2b-it"
</pre><p>use Gemma2</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/100/discussion/#comment-1851447742'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (windmaple) [inline Gemma/Finetune_with_JORA.ipynb:1]: <p>JSOB, typo</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/100/discussion/#comment-1851447743'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

**Comment by windmaple** [comment]
Also does it crash for you on T4? It keeps crashing due to OOM (CPU)

Don't forget to update README/WISHLIST

File: Gemma/Finetune_with_JORA.ipynb (hunk 1 above)
Comment (kinarr) [reply Gemma/Finetune_with_JORA.ipynb:1]: As stated in the library here
https://github.com/aniquetahir/JORA

Comment (kinarr) [reply Gemma/Finetune_with_JORA.ipynb:1]: Not sure if the library is compatible with it but will check

Comment (kinarr) [reply Gemma/Finetune_with_JORA.ipynb:1]: @windmaple The directory structure needs to be modified manually for the time being until JORA's maintainer updates the repo to include Gemma 2 support

**Comment by kinarr** [comment]
> Also does it crash for you on T4? It keeps crashing due to OOM (CPU)
> 
> Don't forget to update README/WISHLIST

I've mentioned in the notebook that it requires an A100 btw.

File: Gemma/Finetune_with_JORA.ipynb (hunk 1 above)
Comment (kinarr) [reply Gemma/Finetune_with_JORA.ipynb:1]: I'll update the notebook to use Gemma 2 but I figured it wouldn't be ideal as the library hasn't added support for it yet. It works nonetheless, but we need to alter the directory structure for the downloaded Flax model.

**Comment by kinarr** [comment]
> Also does it crash for you on T4? It keeps crashing due to OOM (CPU)
> 
> Don't forget to update README/WISHLIST

The library has no quantization support yet so we're expected to use a high-end GPU for the time being I believe.

**Review by JesseyMussche — APPROVED** [review]

**Comment by windmaple** [comment]
> > Also does it crash for you on T4? It keeps crashing due to OOM (CPU)
> > Don't forget to update README/WISHLIST
> 
> I've mentioned in the notebook that it requires an A100 btw.

T4 actually works fine. For some reason you need the high RAM instance. Could you try to figure out why?
Or at least we should point ppl to use Kaggle, which offers more CPU RAM on GPU VMs.

**Comment by kinarr** [comment]
> > > Also does it crash for you on T4? It keeps crashing due to OOM (CPU)
> > > Don't forget to update README/WISHLIST
> > 
> > 
> > I've mentioned in the notebook that it requires an A100 btw.
> 
> T4 actually works fine. For some reason you need the high RAM instance. Could you try to figure out why? Or at least we should point ppl to use Kaggle, which offers more CPU RAM on GPU VMs.

Colab using the latest JAX version might be causing issues with the current JORA implementation. I'll investigate and verify if that's really the case.

**Comment by kinarr** [comment]
Even on Kaggle it's hitting Colab's RAM limits (12.7 GB) so I believe it's best to direct free users to run it on Kaggle? @windmaple 
image

**Comment by windmaple** [comment]
Kaggle it is

**Pull request merged**